pos_product_warning/static/src/js/ProductScreen.esm.js

```diff
@@ -0,0 +1,30 @@
+/** @odoo-module **/
+
+import ProductScreen from "point_of_sale.ProductScreen";
+import Registries from "point_of_sale.Registries";
+
+export const POSProductWarning = (ProductScreen) =>
+    class extends ProductScreen {
+        async _onClickPay() {
+            const order = this.env.pos.get_order();
+            const orderlines = order
+                .get_orderlines()
+                .filter((line) => line.product.pos_warn_msg && line.quantity >= 0);
+            const productSaleWarningList = _.uniq(
+                _.pluck(_.pluck(orderlines, "product"), "pos_warn_msg")
+            );
+            if (productSaleWarningList.length > 0) {
+                const {confirmed} = await this.showPopup("ProductWarningPopup", {
```

Comment:
Can we not use 'SelectionPopup' instead of creating a new popup?

Reply:
> Can we not use 'SelectionPopup' instead of creating a new popup?

SelectionPOPUP is used for the choice of item but in our case, it is a confirmation warning not choice of items that's why we create a new popup.

To confirm a message in SelectionPOPUP, you need to click on one of the message lines.

![image](https://github.com/OCA/pos/assets/20418904/cd533c22-004a-4e6a-8cfb-10b8eb74eda8)

Reply:
> Can we not use 'SelectionPopup' instead of creating a new popup?

SelectionPOPUP is used for the choice of item but in our case, it is a confirmation warning not choice of items that's why we create a new popup.

To confirm a message in SelectionPOPUP, you need to click on one of the message lines.

![image](https://github.com/OCA/pos/assets/77659554/fc485035-4d77-45b8-b9f8-c689635561f8)